src/app/Shared/Services/Report.service.tsx

```diff
@@ -101,13 +101,102 @@ export class ReportService {
     );
   }
 
+  reportJson(recording: Recording, connectUrl: string): Observable<[string, RuleEvaluation][]> {
+    if (!recording?.reportUrl) {
```

Comment:
recording is specified as non-null or non-undefined so no need for `?` here, right?